datafusion/src/physical_plan/hash_join.rs

```diff
@@ -708,7 +720,6 @@ macro_rules! equal_rows_elem {
         let right_array = $r.as_any().downcast_ref::<$array_type>().unwrap();
 
         match (left_array.is_null($left), left_array.is_null($right)) {
-            (true, true) => true,
```

Comment:
this was the line, right? Damm, so many wrong things for this one.

Really great finding, @Dandandan !

Reply:
Yep! To be fair I also added this line myself 😆

I think we also should start testing DataFusion more rigorously wrt null handling, but one step at a time.

Reply:
Even if it is, I noticed the values method doesn't take offset into account right now, so this is maybe an optimization worth doing once it is in the arrow kernel.

Reply:
The fact we are now worrying about and fixing null handling is a sign, in my mind, of DataFusion's maturation 🤣